src/compose/table.c

```diff
@@ -176,14 +175,16 @@ xkb_compose_table_new_from_locale(struct xkb_context *ctx,
     if (!table)
         return NULL;
 
-    cpath = get_xcomposefile_path();
-    if (cpath) {
-        file = fopen(cpath, "rb");
+    path = get_xcomposefile_path();
+    if (path) {
+        file = fopen(path, "rb");
         if (file)
             goto found_path;
     }
+    free(path);
+    path = NULL;
```

Comment:
nitpick: `path = NULL` is superfluous now since you're setting it in the next line anyway. Same in line 194 and 203.

Reply:
212 too, since we then `return NULL;` and don’t use `path` for anything in-between.